static/js/message_fetch.js

```diff
@@ -52,6 +52,10 @@ function process_result(data, opts) {
     stream_list.update_streams_sidebar();
     pm_list.update_private_messages();
 
+    if (!stream_list.is_scrolling) {
```

Comment:
We should try to avoid sharing boolean values between two modules.  If it's necessary to share some state, you can wrap it with a function to encapsulate it better.  But in this case, it's easier to just have a function called `stream_list.maybe_scroll_narrow_into_view` that handles the `is_scrolling` case internally.  We don't want the `message_fetch` modules to be managing scrolling at all beyond a one-line callback to `stream_list`--let stream list handle that.